management_app/views/settings.py

```diff
@@ -1,37 +1,90 @@
 @login_required
-def admins():
-    # TODO get data from database
-    admins = [
-        {'name': 'Admin A', 'email': 'profA@uci.edu'},
-        {'name': 'Staff B', 'email': 'staffB@uci.edu'},
-        {'name': 'Staff C', 'email': 'staffC@uci.edu'}
-    ]
-    users = [
-        {'name': 'User A'},
-        {'name': 'User B'},
-        {'name': 'User C'}
-    ]
+def index():
+    if request.method == 'POST':
+        add_admin_status(request.form['user-select'])
+
+    users, admins = get_users_and_admins()
+    current_user = next(admin for admin in admins if admin['user_email'] == session['email'])
+
     return render_template(
-        'settings/admins.html',
+        'settings/index.html',
         admins=admins,
-        current_user=admins[0],
+        current_user=current_user,
         users=users
     )
 
+@settings.route('/admins/<id>', methods=['POST', 'DELETE'])
+@login_required
+def remove_admin(id):
+    if is_only_admin():
+        flash('Cannot remove admin when only one is left.', 'error')
+        return make_response('Cannot remove admin when only one is left.', 400)
+    if request.method == 'POST':
+        remove_admin_status(id)
+    if request.method == 'DELETE':
+        delete_admin(id)
+    return Response(status=200)
+    
+
 @settings.route('/point-policy')
 @login_required
 def point_policy():
-    # TODO get data from database
-    rules = [
-        {'name': 'Rule 1', 'point_value': 1},
-        {'name': 'Rule 2', 'point_value': 0.5},
-        {'name': 'Rule 3', 'point_value': 1.5}
-    ]
-    return render_template('settings/point-policy.html', rules=rules)
\ No newline at end of file
+    db = get_db()
+    rules = db.execute('SELECT * FROM rules').fetchall()
+    return render_template('settings/point-policy.html', rules=rules)
+
+@settings.route('/point-policy/rules/<id>', methods=['POST'])
+@login_required
+def edit_rule_point_value(id):
+    update_rule_point_value(id, request.form['point-value'])
+    return redirect(url_for('settings.point_policy'))
+
+def get_users_and_admins():
+    db = get_db()
+    admins = []
+    users = []
+    res = db.execute("""
+            SELECT u.user_id, u.user_name, u.user_email, u.user_ucinetid, u.admin, fs.role
+            FROM users AS u LEFT JOIN faculty_status AS fs ON u.user_id = fs.user_id
+            WHERE (fs.end_year IS NULL AND fs.active_status IS TRUE) OR fs.role IS NULL
+        """)
+    for user in res.fetchall():
+        if user['admin']:
+            admins.append(user)
+        else:
+            users.append(user)
+
+    return (users, admins)
+
+def add_admin_status(id):
+    db = get_db()
+    db.execute('UPDATE users SET admin = 1 WHERE user_id = ?', (id))
+    db.commit()
+
+def remove_admin_status(id):
+    db = get_db()
+    db.execute('UPDATE users SET admin = 0 WHERE user_id = ?', (id))
+    db.commit()
+
+def delete_admin(id):
+    db = get_db()
+    db.execute('DELETE FROM users WHERE user_id = ?', (id))
+    db.commit()
```

Comment:
For those SQL statements using variables, you may need to add a comma and make your parameters a tuple since there would be an error of `Incorrect number of bindings supplied.`

Reference: https://stackoverflow.com/questions/16856647/sqlite3-programmingerror-incorrect-number-of-bindings-supplied-the-current-sta

Reply:
Updated and ready for review.